cloud_testing/xctest_parser/src/test/kotlin/xctest/TestArtifact.kt

```diff
@@ -0,0 +1,95 @@
+package xctest
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.io.File
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+abstract class TestArtifact {
+    internal val fixturesPath = "./src/test/kotlin/xctest/fixtures"
+
+    init {
+        val assetLink = remoteAssetLink()
+        if (updateRequired(assetLink)) {
+            updateFixtures(assetLink)
+        }
+    }
+
+    private fun getHtml(): Document {
+        val githubUrl = "https://github.com/Flank/test_artifacts/releases/latest"
+        val maxRetries = 6
```

Comment:
I think it'd be great to abstract the retry code so we're not duplicating the same code for `getHtml()` and `downloadAsset`

wait is only ever used in the retry code. Maybe create a new RetryErrors class.

Reply:
The `downloadAsset` method is no longer needed now that retries are built into the `curl` command.

Reply:
I still think the class refactor makes sense so that there's a path forward for code reuse that doesn't involve copy and paste